osutil/fips/fips_test.go

```diff
@@ -0,0 +1,129 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package fips_test
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/osutil/fips"
+)
+
+func TestFIPSIsEnabled(t *testing.T) {
+	t.Cleanup(func() { dirs.SetRootDir("") })
+
+	mockFipsEnabledWithContent := func(root, content string) {
+		f := filepath.Join(root, "/proc/sys/crypto/fips_enabled")
+		if err := os.MkdirAll(filepath.Dir(f), 0755); err != nil {
+			t.Fatal("cannot create directory", err)
+		}
+
+		if err := os.WriteFile(f, []byte(content), 0444); err != nil {
+			t.Fatal("cannot write file", err)
+		}
+	}
+
+	t.Run("enabled", func(t *testing.T) {
+		dirs.SetRootDir(t.TempDir())
+
+		mockFipsEnabledWithContent(dirs.GlobalRootDir, "1\n")
+
+		res, err := fips.IsEnabled()
+		if err != nil {
+			t.Fatal("unexpected error", err)
+		}
+		if res != true {
+			t.Fatal("expected fips to be enabled")
+		}
```

Comment:
For example, I think in this specific case, this might be nicer with `check`
```suggestion
		res, err := fips.IsEnabled()
		c.Assert(err, IsNil)
		c.Assert(res, Equals, true)
```